README.md

```diff
@@ -44,30 +40,39 @@ Writing tests is simply a case of implementing the types and registering them wi
 package main
 
 import (
-    "github.com/form3tech-oss/f1/v2/pkg/f1"
-    "github.com/form3tech-oss/f1/v2/pkg/f1/testing"
-    "fmt"
+	"fmt"
+	
+	"github.com/form3tech-oss/f1/v2/pkg/f1"
+	"github.com/form3tech-oss/f1/v2/pkg/f1/testing"
 )
 
 func main() {
-    testing.Add("mySuperFastLoadTest", setupMySuperFastLoadTest)
-    f1.Execute()
+	f1.Scenarios().Add("mySuperFastLoadTest", setupMySuperFastLoadTest).Execute()
 }
 
-func setupMySuperFastLoadTest(t *testing.T) (testing.RunFn, testing.TeardownFn) {
-    runFn := func(t *testing.T) {
-        fmt.Println("Wow, super fast!")
-    }
-
-    teardownFn := func(t *testing.T) {
-        fmt.Println("Wow, that was fast!")
-    }
-
-    return runFn, teardownFn
+func setupMySuperFastLoadTest(t *testing.T) testing.RunFn {
+	teardownScenarioFn := func() {
+		fmt.Println("Wow, that was fast! Now let's clean up the scenario")
+	}
+	// Register clean up function which will run at the end of the scenario execution
+    t.Cleanup(teardownScenarioFn)
+
+	teardownIterationFn := func() {
+		fmt.Println("Wow, that was fast! Now let's clean up the iteration")
+	}
+	
+	runFn := func(t *testing.T) {
+		fmt.Println("Wow, super fast!")
+
+		// Register clean up functions which will run in LIFO order after each iteration
+	    t.Cleanup(teardownIterationFn)
+	}
+
+	return runFn
```

Comment:
Use either tabs or spaces

Reply:
I think I'd prefer to see the teardown function inline in the cleanup. The iteration teardown in particular looks strange as it isn't in the context of the run function. 

Perhaps we should also add some comments about where the actual scenario implementation would be done?